python/cuml/cuml/svm/svm_base.pyx

```diff
@@ -222,7 +225,131 @@ class SVMBase(UniversalBase,
+            "cache_size": cache_size,
+            "max_iter": model.max_iter,
+        }
+
+    def _params_to_cpu(self):
+        if (cache_size := self.cache_size) == 1024:
+            # XXX: the cache sizes differ between cuml and sklearn, for now we
+            # just adjust when the value's match the defaults.
+            cache_size = 200
+
+        return {
+            "kernel": self.kernel,
+            "degree": self.degree,
+            "gamma": self.gamma,
+            "coef0": self.coef0,
+            "tol": self.tol,
+            "C": self.C,
+            "cache_size": cache_size,
+            "max_iter": self.max_iter,
+        }
+
+    def _attrs_from_cpu(self, model):
+        if model._sparse:
+            # sklearn stores dual_coef_ and support_vectors_ as sparse
+            # csr_matrix objects when fit on sparse data. cuml always
+            # stores them as dense.
+            dual_coef_ = model.dual_coef_.toarray()
+            support_vectors_ = model.support_vectors_.toarray()
+        else:
+            dual_coef_ = model.dual_coef_
+            support_vectors_ = model.support_vectors_
+
+        return {
+            "dual_coef_": to_gpu(dual_coef_, order="F"),
+            "intercept_": to_gpu(model.intercept_, order="F"),
+            "n_support_": int(model.n_support_.sum()),
+            "support_": to_gpu(model.support_, order="F"),
+            "support_vectors_": to_gpu(support_vectors_, order="F"),
+            "_gamma": float(model._gamma),
+            "_sparse": model._sparse,
+            "dtype": np.float64,
+            "target_dtype": np.int64,
+            "n_rows": model.shape_fit_[0],
+            **super()._attrs_from_cpu(model),
+        }
+
+    def _sync_attrs_from_cpu(self, model):
+        super()._sync_attrs_from_cpu(model)
+        # Clear cached coef_
+        self._internal_coef_ = None
+        # Release existing _model (if any)
+        self._dealloc()
+        # Rebuild _model from new attributes
+        self._model = self._get_svm_model()
+
+    def _attrs_to_cpu(self, model):
+        dual_coef_ = to_cpu(self.dual_coef_, order="C", dtype=np.float64)
+        support_vectors_ = to_cpu(self.support_vectors_, order="C", dtype=np.float64)
+        intercept_ = to_cpu(self.intercept_, order="C", dtype=np.float64)
+
+        if self._sparse:
+            # sklearn stores dual_coef_ and support_vectors_ as sparse
+            # csr_matrix objects when fit on sparse data. cuml always
+            # stores them as dense.
+            dual_coef_ = scipy.sparse.csr_matrix(dual_coef_)
+            support_vectors_ = scipy.sparse.csr_matrix(support_vectors_)
+
+        if hasattr(self, "classes_"):
+            # sklearn's binary classification expects inverted
+            # _dual_coef_ and _intercept_.
+            _dual_coef_ = -dual_coef_
+            _intercept_ = -intercept_
+        else:
+            _dual_coef_ = dual_coef_
+            _intercept_ = intercept_
+
+        return {
```

Comment:
There's a bunch of internal state (Note all the `_` prefixed attributes) that we need to setup here on the sklearn side. This works, but feels a bit brittle since it relies on sklearn not changing their internals. What I have here seems fine for now, just something to keep in mind when thinking about maintainability and testing of `cuml.accel` going forward. The linear models were considerably simpler to convert state to/from sklearn than svm.